src/schedule/ll_schedule.c

```diff
@@ -77,17 +87,23 @@ static inline uint64_t ll_get_timer(struct ll_schedule_data *queue)
 
 static inline void ll_set_timer(struct ll_schedule_data *queue)
 {
-	uint64_t ticks;
+	uint64_t ticks_req;
+	uint64_t ticks_set;
 
 	spin_lock(ll_shared_ctx->lock);
 
 	if (atomic_add(&queue->num_ll, 1) == 1)
 		ll_shared_ctx->timers[cpu_get_id()] = &queue->ts->timer;
 
 	if (atomic_add(&ll_shared_ctx->total_num_work, 1) == 1) {
-		ticks = queue_calc_next_timeout(queue, ll_get_timer(queue));
-		ll_shared_ctx->last_tick = ticks;
-		queue->ts->timer_set(&queue->ts->timer, ticks);
+		ticks_req = queue_calc_next_timeout(queue,
```

Comment:
I don't think we need to check it here. It's calculated from the current time, not last_tick.

Reply:
Yes the argument to `queue_calc_next_timeout()` is current time but there is no guarantee the time returned from it will not be past time
`return queue->ticks_per_msec * queue->timeout / 1000 + start;`

In above statement its enough somebody sets `queue->timeout` to incorrect value and we may have problems.

To summarise I would keep this check here.

Reply:
If it's needed then please add some comments to this effect. This way other will understand why we do the check.